enhancements/image-references.md

```diff
@@ -0,0 +1,228 @@
+the manifests deploy and manage. As with OpenShift release payloads, operator
+bundles must refer to images by digest in order to produce reproducible
+installations. A shared process to build operator bundles that replaces image
+references with fully-resolved pull specifications that pin images by digest
+must be built; this process must allow for a number of separate build systems
+to direct how these replacements occur in order to support a full-featured
+build and test strategy.
+
+## Motivation
+
+### Goals
+
+- there is one, canonical, method for building an operator bundle image layer
+  from a directory of manifests
+- building an operator bundle image does not require the use of a container
+  runtime, elevated privileges or any capacities that are not present for 
+  containerized workloads on OpenShift
+- it is as simple for a developer to build a bundle referring to test versions
+  of operand images as it is for a CI or productized build pipleine to create
+  bundle images for publication
+- operator manifest authors dictate the set of image references in manfiests
+  that must be resolved and pinned
+- operator bundle images may be inspected to determine the pull specifications
+  that were used in the creation of the bundle
+
+### Non-Goals
+
+- operator manifest authors must not be required to define the registry from
+  which any individual build system will resolve image references
+- upstream operator manifests must not be required to know how common names or
+  referecnes change when built in a downstream pipeline
+- no prescriptive statement is made about the specific format or contents of the
+  bundle image layer; any will be transparently supported as an output
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an author of a manifest, I would like to check in manifests to my upstream
+repository that are self-consistent, valid and make no assumptions about the
+build system that will eventually create a bundle image with them.
+
+#### Story 2
+
+As an author of an operand, I would like to create a bundle locally in order
+to test my operator end-to-end on a cluster of my choosing without having to
+edit the core configuration for the operator.
+
+#### Story 3
+
+As an author of a build system, I would like to operate with tooling that allows
+me to clearly define the source of truth for image digests in order to keep the
+build-system-specific configuration to an absolute minimum.
+
+#### Story 4
+
+As an engineer involved in publishing an optional operator, I would like to
+configure semantically equivalent image pull specifications once, in order to not
+need to configure each build system independently.
+
+### Implementation Details/Notes/Constraints
+
+The core problems that must be solved in the implementation of this proposal have
+already been handled in the workflow used in `oc adm release new`. When implementing
+improvements to the `operator-sdk bundle create` process we will simply need to
```

Comment:
What part of `operator-sdk bundle create` does this proposal plan on improving? If it is the metadata generation part, this should be updated to
```suggestion
improvements to the `operator-sdk generate bundle` process we will simply need to
```

Reply:
Sorry, this is changing too fast to keep up with ...